tutorials/slack_integration.md

```diff
@@ -0,0 +1,193 @@
+<a name="slack-events"></a>
+# Integrations - Slack Events
+
+## Table of Contents
+
+* [Credential Storage](#credentials)
+* [Creating endpoint](#endpoint)
+* [Slack Setup](#setup)
+* [Setting Up a Webserver](#webserver)
+* [Conclusion](#conclusion)
+
+### Overview
+
+This tutorial will go over how to setup a simple integration with Slack that allows us to send email using a slack slash command. You'll need the following to complete this tutorial.
+
+* SendGrid API Key
+* Slack OAuth Token
+* Slack account & Valid Slack Workspace
+* ngrok (if you do not have an publicly accessible web endpoint)
+* Sinatra 
+* Sendgrid ruby gem
+
+Slack allows us to create [apps](https://api.slack.com/slack-apps). Apps can have different types of capabilities. One such capability is a [slash command](https://api.slack.com/slash-commands). Slash Commands enable Slack users to interact with your app directly from Slack. We will write a simple slash command with following structure.
+```
+/sendmail /to <comma separated list of email ids> /subject <subject message> /body <Plain text body>
+``` 
+
+
+<a name="credentials"></a>
+### Credential Storage
+
+It's not a good practice to directly insert API keys (or any authorization information) into your script for security reasons. As an alternative, please save your credentials in their respective `.env` files, with the following command:
+```
+echo "export SENDGRID_API_KEY='YOUR_API_KEY'" > sendgrid.env
+echo "export SLACK_SENDMAIL_TOKEN='YOUR_APP_TOKEN'" > slack.env //You will receive this after creating an app.
+source ./sendgrid.env && source ./slack.env
+```
+
+
+<a name="endpoint"></a>
+### Creating endpoint
+Every Slack slash commands needs a publically available endpoint at the time of creating new slash command. Since we are using a development environment, we can use [ngrok](https://ngrok.com/) to expose a locally available endpoint.
```

Comment:
Really nitpicky, I know, but

> publically

should be: 

    publicly

And that ngrok link looks pretty small, too. Maybe instead of:

> we can use [ngrok] to expose a locally available endpoint

We could do:

    we can [use ngrok to expose a locally available endpoint]